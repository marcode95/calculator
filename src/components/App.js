import '../App.css';
import Display from './Display';
import ButtonPanel from './ButtonPanel';

const App = () => { // eslint-disable-line
  return (
    <div className="App">
      <Display />
      <ButtonPanel />
    </div>
  );
};

export default App;
