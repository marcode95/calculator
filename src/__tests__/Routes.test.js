import renderer from 'react-test-renderer';
import Route from '../components/routes';
import Quote from '../routes/quote';
import MainPage from '../routes/mainpage';
import App from '../components/App';

describe('checks if routes are working', () => {
  test('should render to mainpage', () => {
    const tree = renderer
      .create(<Route path="/" component={MainPage} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render app with calculator', () => {
    const tree = renderer
      .create(<Route path="/calculator" component={App} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render quote of the day page', () => {
    const tree = renderer
      .create(<Route path="/calculator" component={Quote} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
