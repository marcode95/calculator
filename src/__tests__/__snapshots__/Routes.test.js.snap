// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if routes are working should render app with calculator 1`] = `
Array [
  <nav>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Main Page
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote of the day
        </a>
      </li>
    </ul>
  </nav>,
  <div>
    <h1>
      Math Magicians
    </h1>
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit.
    </p>
  </div>,
]
`;

exports[`checks if routes are working should render quote of the day page 1`] = `
Array [
  <nav>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Main Page
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote of the day
        </a>
      </li>
    </ul>
  </nav>,
  <div>
    <h1>
      Math Magicians
    </h1>
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit.
    </p>
  </div>,
]
`;

exports[`checks if routes are working should render to mainpage 1`] = `
Array [
  <nav>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Main Page
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote of the day
        </a>
      </li>
    </ul>
  </nav>,
  <div>
    <h1>
      Math Magicians
    </h1>
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit.
    </p>
  </div>,
]
`;
